File: Java/ROUND_2_B/src/main/java/FootballScoreStats.java

public class FootballScoreStats {

	private final FootballData footballData;

	public FootballScoreStats(FootballData footballData) {
		this.footballData = footballData;
	}

	public int teamTotal(String teamName) {
		int total = 0;
		Game[] played = footballData.getAllPlayed();
		for (Game game : played) {
			total += game.getTeamScore(teamName);
		}
		return total;
	}

}

File: Kotlin/ROUND_2_B/src/main/kotlin/FootballScoreStats.kt

class FootballScoreStats(private val footballData: GameData) {
    fun teamTotal(teamName: String): Int {
        var total = 0
        val played: List<Game> = footballData.allPlayed
        for (game in played) {
            total = game.teamScore(teamName, total)
        }
        return total
    }

}

